relayer-engine/src/listener/eventHarness.ts

```diff
@@ -67,9 +67,13 @@ export async function consumeEventHarness(
       logger().info(
         `Received workflow data from plugin ${plugin.pluginName}, adding workflow...`,
       );
+      const hashFirstDigits = parsedVaa?.hash.toString("hex").substring(0, 5);
+      const emitterAddress = parsedVaa?.emitterAddress.toString("hex");
       await storage.addWorkflow({
         data: workflowData,
-        id: hash,
+        id: parsedVaa
+          ? `${parsedVaa.emitterChain}/${emitterAddress}/${parsedVaa.sequence}/${hashFirstDigits}`
```

Comment:
Just curious... how come `parsedVaa` is not required here? sounds like a key piece of data...

Reply:
Yeah actually why is this optional? Higher in the function is appears to be definitely defined

Reply:
you can trigger a workflow with your own custom event that's not necessarily a vaa, but we need to have a follow up discussion on that.

Reply:
I think we mean just mechanically in this file, shouldn't the var parsedVaa always be defined? This is not a blocker, just wondering

Reply:
it will be, but I wasn't sure how you were doing it in the generic relayer if you trigger a custom event where you didn't have a specific vaa but instead a batch that came in the additional data param. It being always defined certainly makes it's a lot easier.